### PR TITLE
Fix some absence of port selection menu

### PR DESCRIPTION
**Pull request by ASDosjani**
I made a fix for `virtual connection` and `show all serial device` so there is no need to restart the configurator if you change these settings from now.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by McGiverGim — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Comment by McGiverGim** [comment]
You don't have to merge master into your branch. To put your branch up to date, you must **rebase** master onto your branch and after that do a `push` with `--force-with-lease`

**Comment by ASDosjani** [comment]
> 

I had to resolve a conflict. Can you help me on DC?

**Comment by haslinghuis** [comment]
Please also verify ELRS wifi is still working.

**Comment by ASDosjani** [comment]
> Please also verify ELRS wifi is still working.

It's working, only the auto mdns is broken. It's shows up for a few seconds but than it dissapeare. It has the same behavior on the latest nightly build so it's not related to my code.

File: src/js/port_handler.js
Comment (haslinghuis) [inline src/js/port_handler.js:21]:
```suggestion
    this.usbCheckLoop = false;
```

File: src/js/port_handler.js
Comment (haslinghuis) [inline src/js/port_handler.js:125]:
```suggestion
    this.initialPorts = false;
```

File: src/js/tabs/firmware_flasher.js
Comment (haslinghuis) [inline src/js/tabs/firmware_flasher.js:19]: This should be removed as there is already a mechanism in place for detection

File: src/js/port_handler.js
Comment (haslinghuis) [inline src/js/port_handler.js:171]: This should be removed as there is already a mechanism in place for detection

File: src/js/port_handler.js
Comment (haslinghuis) [inline src/js/port_handler.js:130]: This should be removed as there is already a mechanism in place for detection

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: src/js/port_handler.js
Comment (ASDosjani) [reply src/js/port_handler.js:130]: This it the point of my PR. The existing mechanism is not sufficient to dynamicall change the ui. I have to restart the configurator for the changes to take place, like on virtual fc or show all serial device switch change. I moved these lines from the PortHandler.initialize functio. On the firmware flasher page this reinitialize functuin should be callled because you don't want to flash a virtual fc... If you know what I want to say. That's why I made this function and to change dinamically if I change a switch on the options tab.

Comment (haslinghuis) [reply src/js/port_handler.js:130]: Have tested my suggestions here and can toggle virtual mode without any problem.

Let's tackle the not flashing a virtual fc in another PR as we just should remove the ability in another way as for now nothing happens if you press flash. We are also going to move to a new API as we need this to be stable.

Comment (ASDosjani) [reply src/js/port_handler.js:130]: I don't want to flash the virtual FC. It just select the detected board if you click on the flash firmware tab.

Comment (haslinghuis) [reply src/js/port_handler.js:130]: Yes I appreciate the last part as having tested your PR (with and without my requested changes).

There was a reason we didn't preselect the board on flasher tab as we had to remove that after a lot of testing with older boards, using DFU boot, legacy flashing, switching between FC's and more.

Comment (ASDosjani) [reply src/js/port_handler.js:130]: Alright, I removed.

File: src/js/tabs/firmware_flasher.js
Comment (ASDosjani) [reply src/js/tabs/firmware_flasher.js]: Done

File: src/js/port_handler.js
Comment (haslinghuis) [reply src/js/port_handler.js:130]: Yes please also remove line 130 and 171 as the `[trigger(']change[")]` function applies only to firmware tab anyway.

This PR focus is resolving the need for reboot changing show all serial devices and virtual flight control (as I just moved two lines in the past to improve performance as Port Handler is in a 500ms loop).

Other changes should go into another PR if there is any need to fix it.

Comment (ASDosjani) [reply src/js/port_handler.js:130]: Are these changes unnecessary or they should go to another PR?

Comment (haslinghuis) [reply src/js/port_handler.js:130]: How to reproduce the bug?

Comment (ASDosjani) [reply src/js/port_handler.js:130]: You just open the configurator and you wont see the msp box or the manual port input box.
image
image

On the left side.

Comment (ASDosjani) [reply src/js/port_handler.js:130]: > How to reproduce the bug? Yes needs another BUGFIX PR if needed.

So there is no bugfix and feature PR at the same time. It makes sense, I just fixed it since it's a single line of code.

Comment (ASDosjani) [reply src/js/port_handler.js]: Removed

**Review by haslinghuis — APPROVED** [review]

**Review by limonspb — DISMISSED** [review]

File: src/js/port_handler.js
Comment (haslinghuis) [inline src/js/port_handler.js:126]:
```suggestion
    clearTimeout(this.usbCheckLoop);
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: src/js/port_handler.js
Comment (ASDosjani) [reply src/js/port_handler.js:126]: Sorry, I missed it when renamed the variable.

**Review by haslinghuis — DISMISSED** [review]

**Review by McGiverGim — DISMISSED** [review]

**Review by haslinghuis — DISMISSED** [review]

**Comment by ASDosjani** [comment]
I changed the timeout cleaning. Can you guys check?

**Review by haslinghuis — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**
